src/conf_mode/system-ipv6.py

```diff
@@ -69,34 +70,42 @@ def generate(ip_opt):
 def apply(ip_opt):
     # disable IPv6 address assignment
     if ip_opt['disable_addr_assignment']:
+        # this file is useless, but it's an indicator for grub setup scripts
         with open(ipv6_disable_file, 'w') as f:
             f.write('options ipv6 disable_ipv6=1')
+
+        if os.path.exists(grub_cfg_file):
+            os.system("sed -i 's/ipv6.disable=[^[:space:]]*/ipv6.disable=1/;/^[[:space:]]*linux / {/ipv6.disable=/! s/$/ ipv6.disable=1/}' "+grub_cfg_file)
```

Comment:
I do not know if it is the right way to offload this to `os.system()` or rather parse it via means of Python. @dmbaturin ?

Reply:
I like the patch in general and the idea - this is a minor detail we can improve later on.

Reply:
i'm not a python coder. i found this simple. if there is a robust/community standard solution for replace sed in python, good for me.